common/environment/environment.cpp

```diff
@@ -1606,25 +1598,48 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
         addPathSepChar(fullDropZoneDir);
         const char * pdropzoneDir = fullDropZoneDir.str();
 
-        // Check target file path starts with this Drop zone path
-        // the drop zone paths can be nested (nothing forbids it) like
```

Comment:
More or less, yes. That was an approach where we have one set of Drop Zones and each assigned only one machine. Now we can have a set of Drop Zones and set of machines for each zones. I think there is no real sense to search a shortest matching drop zone because it should match the server as well. So now if the drop zone directory match to the first part of target filename, then it checks computer (if it is exists - backward compatibility) or the server list. If there is a match then the operation gets green light. If the server doesn't match to any of the server list then the check (outside loop) goes to the next drop zone.